### PR TITLE
Added IntMaskType, IntMaskOfType

### PR DESCRIPTION
**Pull request by vudaltsov**

### CODE REVIEW

File: tests/psalm/IntMaskOfTest.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+namespace Typhoon\Type\IntMaskOfTest;
+
+use Typhoon\Type\IntMaskOfType;
+use Typhoon\Type\Type;
+use function Typhoon\Type\extractType;
+
+/**
+ * @param Type<\ReflectionClass::IS_*> $constantType
+ */
+function a(Type $constantType): void
```
Comment (klimick) [inline tests/psalm/IntMaskOfTest.php]: It would be great to add one more test for `IntMaskType`.

Comment (vudaltsov) [reply tests/psalm/IntMaskOfTest.php]: Done!

**Review by klimick — APPROVED** [review]

File: src/types.php [hunk 2]
```diff
@@ -81,6 +83,28 @@ public static function positiveInt(): IntRangeType
         return new IntRangeType(1);
     }
 
+    /**
+     * @psalm-pure
+     * @no-named-arguments
+     * @param positive-int $int
+     * @param positive-int ...$ints
```
Comment (aivchen) [inline src/types.php]: Here the type of parameters is `positive-int`. But `IntMaskType` accepts `non-negative-int`. Is this intentional?

Comment (vudaltsov) [reply src/types.php]: Thank you, fixed!

**Pull request merged**